timemachine/cpp/src/kernels/k_flat_bottom_bond.cuh

```diff
@@ -8,7 +8,13 @@ template <typename RealType>
 RealType __device__ __forceinline__ compute_flat_bottom_energy(RealType k, RealType r, RealType rmin, RealType rmax) {
     RealType r_gt_rmax = static_cast<RealType>(r > rmax);
     RealType r_lt_rmin = static_cast<RealType>(r < rmin);
-    return (k / 4) * (r_lt_rmin * (pow(r - rmin, 4)) + r_gt_rmax * (pow(r - rmax, 4)));
+    RealType d_rmin_2 = (r - rmin) * (r - rmin);
```

Comment:
nit: I assume there's no difference after compiler optimization, but any reason not to eliminate common subexpressions here? E.g.

```suggestion
    RealType d_rmin = r - rmin;
    RealType d_rmin_2 = d_rmin * d_rmin;
```